### PR TITLE
Codestarts - add gRPC codestart

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: extensions/grpc/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -8,4 +8,8 @@ metadata:
     - "web"
     - "serialization"
     - "reactive"
-  status: "experimental"
\ No newline at end of file
+  status: "experimental"
+  codestart:
+    name: "grpc"
+    languages:
+      - "java"
```
Comment (ia3andy) [inline extensions/grpc/runtime/src/main/resources/META-INF/quarkus-extension.yaml:15]: you need to add the  artifact: "io.quarkus:quarkus-project-core-extension-codestarts"

**Review by ia3andy — CHANGES_REQUESTED** [review]

**Review by michalszynkiewicz — APPROVED** [review]
gRPC-wise, I think it's good

**Comment by ia3andy** [comment]
Also I don't see the snapshots in the PR, they should be pushed

File: integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/GrpcCodestartTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    @RegisterExtension
+    public static QuarkusCodestartTest codestartTest = QuarkusCodestartTest.builder()
+            .codestarts("grpc")
+            .languages(JAVA)
+            .build();
+
+    @Test
+    void testContent() throws Throwable {
+        codestartTest.checkGeneratedSource("org.acme.HelloGrpcService");
```
Comment (ia3andy) [inline integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/GrpcCodestartTest.java:21]:
```suggestion
        codestartTest.checkGeneratedSource("org.acme.HelloGrpcService");
        codestartTest.assertThatGeneratedFileMatchSnapshot(JAVA, "src/main/proto/hello.proto");
```

@mkouba

**Comment by mkouba** [comment]
> Also I don't see the snapshots in the PR, they should be pushed

TBH I don't understand the concept of "snapshots" yet but sure I'll add them ;-)

**Review by ia3andy — APPROVED** [review]
LGTM thanks a lot @mkouba :)

File: integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto [hunk 3]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "ilove.quark.us";
```
Comment (ebullient) [inline integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto:4]: while super fun, I think I would stick with "org.acme" .. OR, I would do something more standard: `dev.quarkus.grpc` or something. `ilove` is cute, but I don't like this projection of feelings.

**Review by ebullient — DISMISSED** [review]

File: integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto (hunk 3 above)
Comment (ia3andy) [reply integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto:4]: it's just for the test @ebullient, it's the "test" groupId

Comment (ia3andy) [reply integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto:4]: https://github.com/quarkusio/quarkus/blob/main/independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/codestarts/QuarkusCodestartTest.java#L44

Comment (ebullient) [reply integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto:4]: Fair enough. ;)

File: integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+message HelloRequest {
+    string name = 1;
+}
+
+message HelloReply {
+    string message = 1;
+    int32 count = 2;
```
Comment (ia3andy) [inline integration-tests/devtools/src/test/resources/__snapshots__/GrpcCodestartTest/testContent/src_main_proto_hello.proto:19]:
```suggestion
```

File: devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/grpc-codestart/base/src/main/proto/hello.tpl.qute.proto [hunk 5]
```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "{project.package-name}";
```
Comment (ia3andy) [inline devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/grpc-codestart/base/src/main/proto/hello.tpl.qute.proto:4]:
```suggestion
option java_package = "{project.package-name or 'org.acme'}";
```

**Comment by ia3andy** [comment]
@mkouba I added a fix because the project.package-name didn't have a default, I will have a look to make sure it has a default in the input in another PR.

**Comment by ia3andy** [comment]
> @mkouba I added a fix because the project.package-name didn't have a default, I will have a look to make sure it has a default in the input in another PR.

In the end I added the fix in the PR it will be easier to backport

**Comment by ia3andy** [comment]
I am not sure if the gradle error is related, it doesn't seem so..

**Comment by mkouba** [comment]
> I am not sure if the gradle error is related, it doesn't seem so..

I can see a different gradle multi-module windows failure in my other PR as well :-(

**Pull request merged**